yarn-project/simulator/src/avm/opcodes/addressing_mode.ts

```diff
@@ -63,7 +65,11 @@ export class Addressing {
       resolved[i] = offset;
       if (mode & AddressingMode.RELATIVE) {
         mem.checkIsValidMemoryOffsetTag(0);
-        resolved[i] += Number(mem.get(0).toBigInt());
+        const baseAddr = Number(mem.get(0).toBigInt());
+        resolved[i] += baseAddr;
+        if (resolved[i] > maxUint32) {
```

Comment:
Hmm I'm not sure this should be done here. And I mean that twice:
(1) a relative address that is too big is probably not bad until you try to resolve it
(2) in principle I'd move this to the "INDIRECT" if statement but...

These things might be better handled inside mem.get, which is the part of the system knowing about the memory limit. A red flag is that you had to import this maxUint32, but what if we change the memory size?

Also in your followup work, you will need to handle memory accesses in each opcode. I think what you are doing is assuming "if the resolution already 'constrains' the addresses, then they should be right". That is _partially_ ok. But think of the cases where we read slices. Then the slice+length is not sanitized and it falls back to the memory component to validate it. Which it does, with an assert.

Overall, on the simulator side, it might be better to just change the memory assertions to exceptions.